lib/src/analyzer/rules/use_alternative_class/use_alternative_class_visitor.dart

```diff
@@ -0,0 +1,61 @@
+import 'dart:collection';
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer_plugin/protocol/protocol_generated.dart';
+
+import '../../rule/rule_visitor.dart';
+import '../../rule/utils.dart';
+import 'use_alternative_class.dart';
+
+class UseAlternativeClassVisitor extends RecursiveRuleVisitor {
+  UseAlternativeClassVisitor({
+    required super.rule,
+    required super.result,
+    required super.config,
+  }) {
+    final allowedClasses = config.options[allowedClassesKey] as List<Object>?;
+
+    _alternatives =
+        (config.options[alternativeKey] as UnmodifiableMapView<String, Object?>?)
+                ?.entries.where((e) => allowedClasses == null || !allowedClasses.contains(e.key))
+                .map((e) => MapEntry(e.key, e.value! as String)).toList() ?? [];
+  }
+
+  static const alternativeKey = 'alternatives';
+  static const allowedClassesKey = 'allowed_classes';
+
+
+  late final List<MapEntry<String, String>> _alternatives;
```

Comment:
No reason to use a List<MapEntry>.

```suggestion
  late final Map<String, String> _alternatives;
```

Reply:
made changes to use map